src/crypto/public_key/elliptic/eddsa.js

```diff
@@ -68,11 +84,27 @@ export async function sign(algo, hashAlgo, message, publicKey, privateKey, hashe
     throw new Error('Hash algorithm too weak for EdDSA.');
   }
   switch (algo) {
-    case enums.publicKey.ed25519: {
-      const secretKey = util.concatUint8Array([privateKey, publicKey]);
-      const signature = ed25519.sign.detached(hashed, secretKey);
-      return { RS: signature };
-    }
+    case enums.publicKey.ed25519:
+      try {
+        const webCrypto = util.getWebCrypto();
+        const jwk = privateKeyToJWK(algo, publicKey, privateKey);
+        const key = await webCrypto.importKey('jwk', jwk, 'Ed25519', false, ['sign']);
+
+        const signature = new Uint8Array(
+          await webCrypto.sign('Ed25519', key, hashed)
+        );
+
+        return { RS: signature };
+      } catch (err) {
+        // We do not fallback if the error is related to key integrity
+        if (err.name === 'DataError' || err.name === 'OperationError') {
+          throw err;
+        }
+        const secretKey = util.concatUint8Array([privateKey, publicKey]);
+        const signature = ed25519.sign.detached(hashed, secretKey);
+        return { RS: signature };
```

Comment:
Maybe it's safer to do this the other way around, i.e.:
```suggestion
        if (err.name === 'NotSupportedError') {
          const secretKey = util.concatUint8Array([privateKey, publicKey]);
          const signature = ed25519.sign.detached(hashed, secretKey);
          return { RS: signature };
        }
        throw err;
```